src/solvers/smt2/smt2_solver.cpp

```diff
@@ -41,6 +41,22 @@ class smt2_solvert:public smt2_parsert
   std::set<irep_idt> constants_done;
 };
 
+static std::string smt2_string_literal(const std::string &s)
+{
+  // " is the only escape sequence
+  std::string result;
+  result += '"';
+
+  for(const auto &c : s)
+    if(c == '"')
+      result += "\"\"";
+    else
+      result += c;
```

Comment:
How about `replace_all` from string_utils.{h,cpp}?

Reply:
Uh, actually, that's implemented in a really inefficient way when the string gets longer. Let's say separate PR.

Reply:
I find it quite brave that a character-by-character `operator+=` on a string is used in an argument about efficiency. Also I wonder whether anyone uses `echo` with long strings.

Reply:
The loop in smt2_string_literal is quasi linear (assuming a standard allocation strategy), despite the +=.
The one in replace_all is quadratic.

Reply:
When you say "standard allocation strategy" what is it that you are assuming? Growing by some factor > 1? That at least does not seem to be the case with the STL implementation shipped with GCC 8, which just increases the allocation by the size of the string being appended.

Reply:
No, this refers to the ability of the allocator to avoid fragmentation; the threat is that this becomes quadratic if there's a  non-trivial number of re-allocations.

Reply:
I was wrong in my previous comment, `_M_create` actually does exponential growth of the allocation. So we should just fix `replace_all`.